packages/data/src/world/mq/deku_tree_mq.yml

```diff
@@ -2,126 +2,282 @@
-    "Deku Tree Water Room Back": "(is_child && has_shield) || can_longshot || (can_hookshot && has_iron_boots)"
+    "MQ Deku Tree 3rd Floor": "true"
+    "MQ Deku Tree Slingshot Room Exit Door": "true"
+"MQ Deku Tree Slingshot Room Exit Door":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Slingshot Room Entry Door": "event(DEKU_MQ_SLINGSHOT_ENEMIES)"
+    "MQ Deku Tree MQ Slingshot Room": "true"
+"MQ Deku Tree MQ Slingshot Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Slingshot Room Exit Door": "true"
   events:
-    MQ_DEKU_WATER_TORCHES: "(is_child && has_shield && can_use_sticks) || (can_use_bow && can_play_elegy)"
+    DEKU_MQ_SLINGSHOT_ENEMIES: "soul_deku_baba && soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && (has_weapon || can_use_sticks || has_ranged_weapon_child)" #Bombs and Din's work, but it's horrible.
+    DEKU_MQ_SLINGSHOT_TORCH: "has_fire || can_use_sticks"
+  locations:
+    "MQ Deku Tree Slingshot Chest": "event(DEKU_MQ_SLINGSHOT_ENEMIES)"
+    "MQ Deku Tree Slingshot Room Far Chest": "event(DEKU_MQ_SLINGSHOT_TORCH)"
+    "MQ Deku Tree Heart Slingshot Room": "true"
+    "MQ Deku Tree Grass Slingshot Room Front 1": "can_cut_grass"
+    "MQ Deku Tree Grass Slingshot Room Front 2": "can_cut_grass"
+    "MQ Deku Tree Grass Slingshot Room Back 1": "can_cut_grass"
+    "MQ Deku Tree Grass Slingshot Room Back 2": "can_cut_grass"
+"MQ Deku Tree Basement":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Lobby": "true"
+    "MQ Deku Tree Basement Ledge": "is_adult || event(DEKU_BLOCK) || trick(OOT_DEKU_SKIP) || has_hover_boots || hookshot_anywhere || climb_anywhere"
+    "MQ Deku Tree Room Before Water Room Entry Door": "true"
+  events:
+    DEKU_MQ_WATER_PATH_TORCH1: "has_fire || can_use_sticks" #You need to go through the next room with fire sticks for this event. From xeno: "cond(setting.doorShuffle !== none, has_fire, (roomClearEvent && has_sticks) || has_fire)"
+    DEKU_MQ_BASEMENT_EYE_SWITCH: "can_use_bow || can_use_slingshot"
+  locations:
+    "MQ Deku Tree Basement Chest": "can_use_sticks || has_fire"
+    "MQ Deku Tree Grass Basement Lower 1": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 2": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 3": "can_cut_grass"
+    "MQ Deku Tree Grass Basement Lower 4": "can_cut_grass"
+"MQ Deku Tree Room Before Water Room Entry Door":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Basement": "true"
+    "MQ Deku Tree Room Before Water Room Door to Basement": "event(DEKU_MQ_BASEMENT_EYE_SWITCH)"
+"MQ Deku Tree Room Before Water Room Door to Basement":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Water Room Entry Door": "event(DEKU_MQ_ROOM_BEFORE_WATER_CLEAR)"
+    "MQ Deku Tree Room Before Water Room": "true"
+"MQ Deku Tree Room Before Water Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room Before Water Room Door to Basement": "true"
+    "MQ Deku Tree Room Before Water Room Door to Water Room": "true"
+  events:
+    DEKU_MQ_WATER_PATH_TORCH1: "has_fire"
```

Comment:
"cond(setting.doorShuffle !== none, has_fire, (roomClearEvent && has_sticks) || has_fire)"

This is meant to be here, not above. There isn't a reason to have the other event. And should include sticks back into the current logic since door rando is not yet a thing. Doesn't matter which direction you come from either as both directions have a torch you can carry you through.

Reply:
Moved the note.

Sticks is in logic still from the event in the basement room; the events are the same name.

And direction does matter as the Door to Water Room is barred when coming from Water Room; whereas door to Basement is open when clearing the room.